fix Object.freeze() description to match specs (#20263)
mdn/content: files/en-us/web/javascript/reference/global_objects/object/freeze/index.md

---
{{JSRef}}

> The **`Object.freeze()`** method _freezes_ an object. Freezing an object [prevents extensions](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/preventExtensions) and makes existing properties non-writable and non-configurable. A frozen object can no longer be changed: new properties cannot be added, existing properties cannot be removed, their enumerability, configurability, writability, or value cannot be changed unless its descriptor is accessor descriptor and has a setter inside, and the object's prototype cannot be re-assigned. `freeze()` returns the same object that was passed in.

Reviewer: That setter cannot set anything on the object. We have already described it below:

> still give the illusion that you are changing the value

I don't see how this is necessary for the intro.

Author: Okay, but we still can have a way to change the value, by changing its property's writable value to be true, then we can change it even it's not configurable, but for the intro, we can omit that as you've said

## Description

> Freezing an object is equivalent to [preventing extensions](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/preventExtensions) and then changing all existing [properties' descriptors](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty#description) to `configurable: false, writable: false`, unless they are of accessor descriptor flavor it only sets `configurable: false`. Nothing can be added to or removed from the properties set of a frozen object. Any attempt to do so will fail, either silently or by throwing a {{jsxref("TypeError")}} exception (most commonly, but not exclusively, when in {{jsxref("Strict_mode", "strict mode", "", 1)}}).

Reviewer: ```suggestion
Freezing an object is equivalent to [preventing extensions](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/preventExtensions) and then changing all existing [properties' descriptors'](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty#description) `configurable` to `false` — and for data properties, `writable` to `false` as well. Nothing can be added to or removed from the properties set of a frozen object. Any attempt to do so will fail, either silently or by throwing a {{jsxref("TypeError")}} exception (most commonly, but not exclusively, when in {{jsxref("Strict_mode", "strict mode", "", 1)}}).
```

For data properties of a frozen object, their values cannot be changed since the writable and

> configurable attributes are set to false. Accessor properties (getters and setters) work the same (and still give the illusion that you are changing the value).Note that values

Reviewer: ```suggestion
configurable attributes are set to false. Accessor properties (getters and setters) work the same (and the property value returned by the getter may still change). Note that values
```

Author: How about "but the property's value returned by the getter may still change, and the setter still be called without any errors thrown"

Reviewer: Sounds good to me:

```suggestion
configurable attributes are set to false. Accessor properties (getters and setters) work the same — the property value returned by the getter may still change, and the setter can still be called without throwing errors when setting the property. Note that values
```

> configurable attributes are set to false. Accessor properties (getters and setters) behave differently as only configurable is set to false, so their values can be modified if there is a setter for it. Note that values

Reviewer: This is misleading. The setter cannot actually set anything _on_ the object. Sure, it can set the value in some other store and later read it through the getter, but that has nothing to do with "only configurable is set to false" or even freezing in general. That's just because it's a getter and it can do whatever it wants.

```js
Object.freeze({
  get party() {
    return Math.random();
  },
});
```

Author: I know what you mean, but I'm talking about the concept itself, that we have a way to change this property's value, and this breaks the rule here that they can't be ever changed, and you can depend on that in your code.

Author: we can rephrase it to resolve this confusion and be more accurate, but if we can't agree with this, we can only change the fact that this method only changes the configurable property for properties with accessor descriptor
